packages/discord.js/src/client/BaseClient.js

```diff
@@ -14,6 +14,12 @@ class BaseClient extends EventEmitter {
   constructor(options = {}) {
     super({ captureRejections: true });
 
+    if (options.rest?.userAgentAppendix !== undefined) {
```

Comment:
This should be below line 23 so it checks for `options`'s type first. Otherwise passing `null` will result on a different error.